db_processor.bal

```diff
@@ -288,9 +288,16 @@ function deleteAidPackageItem(int packageId, int packageItemId) returns error? {
     int needId = check dbClient->queryRow(`SELECT NEEDID FROM AID_PACKAGE_ITEM WHERE PACKAGEID=${packageId}
                                         AND PACKAGEITEMID=${packageItemId};`);
     _ = check dbClient->execute(`UPDATE MEDICAL_NEED SET NEEDEDQUANTITY = NEEDEDQUANTITY + ${itemQuantity} WHERE NEEDID=${needId};`);
-    _ = check dbClient->execute(`DELETE AID_PACKAGE_ITEM FROM AID_PACKAGE_ITEM INNER JOIN AID_PACKAGE ON
+    sql:ExecutionResult result = check dbClient->execute(`DELETE AID_PACKAGE_ITEM FROM AID_PACKAGE_ITEM INNER JOIN AID_PACKAGE ON
     AID_PACKAGE_ITEM.PACKAGEID=AID_PACKAGE.PACKAGEID WHERE
     AID_PACKAGE_ITEM.PACKAGEID=${packageId} AND PACKAGEITEMID=${packageItemId} AND (STATUS="Draft" OR STATUS="Published");`);
+    if result.affectedRowCount != 1 {
+        AidPackage aidPackage = check getAidPackage(packageId);
+        if aidPackage.status != "Draft" || aidPackage.status != "Published" {
```

Comment:
@manuranga are we expecting to avoid editing items other than they are in `Draft` or `Published` state? 
if so souldn't the logic be `if !(aidPackage.status == "Draft" || aidPackage.status == "Published"` like this.

Reply:
Ahh yes, big bug, thanks for catching it. I only tested negative case.